libbpf-cargo/src/test.rs

```diff
@@ -753,6 +753,92 @@ fn test_skeleton_builder_clang_opts() {
         .unwrap();
 }
 
+#[test]
+fn test_skeleton_builder_arrays_ptrs() {
+    let (_dir, proj_dir, cargo_toml) = setup_temp_project();
+
+    // Add prog dir
+    create_dir(proj_dir.join("src/bpf")).expect("failed to create prog dir");
+
+    // Add a prog
+    let mut prog = OpenOptions::new()
+        .write(true)
+        .create(true)
+        .open(proj_dir.join("src/bpf/prog.bpf.c"))
+        .expect("failed to open prog.bpf.c");
+
+    write!(
+        prog,
+        r#"
+        #include "vmlinux.h"
+        #include <bpf/bpf_helpers.h>
+
+        struct mystruct {{
+            int x;
+        }};
+
+        const volatile struct mystruct my_array[0] = {{ {{0}} }};
```

Comment:
can you please also check that if struct has an array member everything works? Especially if it's an array of structs. Thanks!

Reply:
Added some extra checks.